### DISCUSSION TITLE
Request for 32 bit deadbeef-static .deb file, version 0.7.2

### DISCUSSION THREAD

**Discussion by lumixdeee** [DeaDBeeF-Player/deadbeef#2810] [General]
On sourceforge, the latest 32 bit static version in .deb format is version 0.6x which still has the dependency on gtk2.0

Please can we have this in version 0.7.2

Deadbeef version: 0.7.2
OS: debian linux , usually lmde

**Comment by Oleksiy-Yakovenko** [comment]
We no longer provide 32 bit binaries, and there are no plans to do so.
32 bit support is no longer guaranteed to work, since we are not testing on 32 bit systems.
However, we don't intentionally remove 32 bit support, and there's a chance that you would be able to build on your own.

**Comment by lumixdeee** [comment]
I have no idea how to build my own! Somebody is compiling the regular .deb file in 32 bit for the deb-multimedia.org repositories
They have version 0.7.2 in stretch and 1.8.1 in later repos.
It's a bit of a chore to install that way, much easier to have the static version.
I'm using it in 32 bit all the time. Might be able to do some testing.

**Comment by Oleksiy-Yakovenko** [comment]
> I have no idea how to build my own!

It's as simple as following the instructions written in the README.md file

**Comment by lumixdeee** [comment]
Thanks, so do I download the source code (from where???) into a folder, then make that folder the current working directory, then install git, Clan toolchain then follow the instructions in readme.md?

It looks like I need to feed it some code, as the readme.md instructions seem to be independent of the deadbeef source code?

**Comment by Oleksiy-Yakovenko** [comment]
You don't have to install git, but the rest is required.
Latest source code is here: https://sourceforge.net/projects/deadbeef/files/travis/linux/1.9.1/deadbeef-1.9.1.tar.bz2/download
Unpack, and follow instructions.
There's nothing super unusual -- just the typical linux build steps.
* Install dependencies (including clang toolchain)
* export CC=clang CXX=clang++
* ./configure
* make
* sudo make install

**Comment by lumixdeee** [comment]
where am I going wrong?

root@ausra:~# apt install Clang toolchain
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package Clang
E: Unable to locate package toolchain
root@ausra:~# synaptic

(synaptic:11142): Gtk-CRITICAL **: 23:05:09.907: gtk_widget_hide: assertion 'GTK_IS_WIDGET (widget)' failed
Discarding: 8 over 9
Discarding: 14 over 15
Discarding: 14 over 15
root@ausra:~# apt install git
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  git-man liberror-perl
Suggested packages:
  git-daemon-run | git-daemon-sysvinit git-doc git-el git-email git-gui gitk
  gitweb git-cvs git-mediawiki git-svn
Recommended packages:
  ssh-client
The following NEW packages will be installed:
  git git-man liberror-perl
0 upgraded, 3 newly installed, 0 to remove and 6 not upgraded.
Need to get 7,386 kB of archives.
After this operation, 37.9 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 https://deb.debian.org/debian bullseye/main amd64 liberror-perl all 0.17029-1 [31.0 kB]
Get:2 https://deb.debian.org/debian bullseye/main amd64 git-man all 1:2.30.2-1 [1,827 kB]
Get:3 https://deb.debian.org/debian bullseye/main amd64 git amd64 1:2.30.2-1 [5,527 kB]
Fetched 7,386 kB in 27s (276 kB/s)                                             
Selecting previously unselected package liberror-perl.
(Reading database ... 241238 files and directories currently installed.)
Preparing to unpack .../liberror-perl_0.17029-1_all.deb ...
Unpacking liberror-perl (0.17029-1) ...
Selecting previously unselected package git-man.
Preparing to unpack .../git-man_1%3a2.30.2-1_all.deb ...
Unpacking git-man (1:2.30.2-1) ...
Selecting previously unselected package git.
Preparing to unpack .../git_1%3a2.30.2-1_amd64.deb ...
Unpacking git (1:2.30.2-1) ...
Setting up liberror-perl (0.17029-1) ...
Setting up git-man (1:2.30.2-1) ...
Setting up git (1:2.30.2-1) ...
Processing triggers for man-db (2.9.4-2) ...
localepurge: Disk space freed:   8572 KiB in /usr/share/locale
localepurge: Disk space freed:      0 KiB in /usr/share/man
localepurge: Disk space freed:      0 KiB in /usr/share/gnome/help
localepurge: Disk space freed:      0 KiB in /usr/share/aptitude
localepurge: Disk space freed:      0 KiB in /usr/share/help

Total disk space freed by localepurge: 8572 KiB

root@ausra:~# cd /home/remix/make/deadbeef-0.7.2/
root@ausra:/home/remix/make/deadbeef-0.7.2# git submodule update --init
fatal: not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
root@ausra:/home/remix/make/deadbeef-0.7.2# ./autogen.sh
-bash: ./autogen.sh: No such file or directory
root@ausra:/home/remix/make/deadbeef-0.7.2# nano autogen.sh


Use "fg" to return to nano.

[1]+  Stopped                 nano autogen.sh
root@ausra:/home/remix/make/deadbeef-0.7.2# xed

** (xed:17468): WARNING **: 23:19:49.638: Could not load Peas repository: Typelib file for namespace 'Peas', version '1.0' not found

** (xed:17468): WARNING **: 23:19:49.638: Could not load PeasGtk repository: Typelib file for namespace 'PeasGtk', version '1.0' not found

(xed:17468): Gtk-WARNING **: 23:19:50.616: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.

** (xed:17468): WARNING **: 23:20:22.880: Set document metadata failed: Setting attribute metadata::xed-spell-language not supported

** (xed:17468): WARNING **: 23:20:22.881: Set document metadata failed: Setting attribute metadata::xed-encoding not supported

** (xed:17468): WARNING **: 23:20:28.535: Set document metadata failed: Setting attribute metadata::xed-position not supported
root@ausra:/home/remix/make/deadbeef-0.7.2# autogen.sh
autogen.sh: command not found
root@ausra:/home/remix/make/deadbeef-0.7.2# ls
autogen.sh  deadbeef  deadbeef.png  doc  locale  pixmaps  plugins
root@ausra:/home/remix/make/deadbeef-0.7.2# nemo

(nemo:17523): Gtk-WARNING **: 23:21:16.300: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
ERROR SET META:15 Setting attribute metadata::nemo-icon-view-auto-layout not supported
CinnamonDesktop-Message: 23:21:16.700: Ignoring thumbnailer with missing binary: '/usr/bin/gdk-pixbuf-thumbnailer'
Running autopoint...
/home/remix/make/deadbeef-0.7.2/autogen.sh: 4: autopoint: not found
Running aclocal...
/home/remix/make/deadbeef-0.7.2/autogen.sh: 5: aclocal: not found
root@ausra:/home/remix/make/deadbeef-0.7.2# autogen.sh
autogen.sh: command not found
root@ausra:/home/remix/make/deadbeef-0.7.2# xed configure.ac

** (xed:17573): WARNING **: 23:25:18.960: Could not load Peas repository: Typelib file for namespace 'Peas', version '1.0' not found

** (xed:17573): WARNING **: 23:25:18.960: Could not load PeasGtk repository: Typelib file for namespace 'PeasGtk', version '1.0' not found

(xed:17573): Gtk-WARNING **: 23:25:19.900: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.

** (xed:17573): WARNING **: 23:25:31.255: Set document metadata failed: Setting attribute metadata::xed-spell-language not supported

** (xed:17573): WARNING **: 23:25:31.255: Set document metadata failed: Setting attribute metadata::xed-encoding not supported

** (xed:17573): WARNING **: 23:25:36.716: Set document metadata failed: Setting attribute metadata::xed-position not supported
root@ausra:/home/remix/make/deadbeef-0.7.2# co
codepage                  compare-im6               compute_member
col                       compare-im6.q16           compute_relabel
colcrt                    compgen                   conjure
colrm                     complete                  conjure-im6
column                    compopt                   conjure-im6.q16
combinediff               compose                   continue
comm                      composite                 convert
command                   composite-im6             convert-im6
command-not-found         composite-im6.q16         convert-im6.q16
command_not_found_handle  compute_av                coproc
compare                   compute_create            corelist
root@ausra:/home/remix/make/deadbeef-0.7.2# nemo

(nemo:17582): Gtk-WARNING **: 23:25:56.281: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
ERROR SET META:15 Setting attribute metadata::nemo-icon-view-auto-layout not supported
CinnamonDesktop-Message: 23:25:56.711: Ignoring thumbnailer with missing binary: '/usr/bin/gdk-pixbuf-thumbnailer'
# _g_io_module_get_default: Found default implementation dconf (DConfSettingsBackend) for ‘gsettings-backend’
# watch_fast: "/org/gnome/terminal/legacy/" (establishing: 0, active: 0)
# unwatch_fast: "/org/gnome/terminal/legacy/" (active: 0, establishing: 1)
# watch_established: "/org/gnome/terminal/legacy/" (establishing: 0)
root@ausra:/home/remix/make/deadbeef-0.7.2# con
conjure          conjure-im6.q16  convert          convert-im6.q16
conjure-im6      continue         convert-im6      
root@ausra:/home/remix/make/deadbeef-0.7.2# configure.ac
configure.ac: command not found
root@ausra:/home/remix/make/deadbeef-0.7.2# CC=clang CXX=lang++ configure
configure: command not found
root@ausra:/home/remix/make/deadbeef-0.7.2# CC=clang CXX=lang++
root@ausra:/home/remix/make/deadbeef-0.7.2# make install
make: *** No rule to make target 'install'.  Stop.
root@ausra:/home/remix/make/deadbeef-0.7.2#

**Comment by Oleksiy-Yakovenko** [comment]
You don't seem to be following instructions precisely enough.
Please try again, and note that commands need to be entered precisely.
From what I can tell,  you typed `configure` instead of `./configure`.
Here's a screenshot of the instructions from README.md

<img width="861" alt="image" src="https://user-images.githubusercontent.com/1895698/171295199-cb0db7d9-a198-43d0-88e1-53a955a8498c.png">

**Comment by lumixdeee** [comment]
`root@ausra:/# ./configure
./configure: line 1: dnl: command not found
./configure: line 2: syntax error near unexpected token `[deadbeef],'
./configure: line 2: `AC_INIT([deadbeef], [devel])'
root@ausra:/# `

Tried to create 'configure' by copying the text from configure.ac from this website and pasting it into xed.

**Comment by Oleksiy-Yakovenko** [comment]
you can't create the `configure` file by renaming `configure.ac`.
What you really need is:
* unpack [the tarball](https://sourceforge.net/projects/deadbeef/files/Archive/deadbeef-0.7.2.tar.bz2/download)
* cd to the unpacked folder
* it will have the configure script already generated
* follow the instructions from INSTALL file (or from the README.md)
* which basically means:
* ./configure
* make
* sudo make install

**Comment by artenax** [comment]
Building Deadbeef from source code is easy enough. The hardest part is installing the build dependencies and compilers. Install Deadbeef from deb-multimedia, run sudo apt build-dep deadbeef-dmo and then you can remove and build from source.
Here are my instructions. ./configure --help will show you a list of available build options, although you don't have to change them and can rely on the defaults.
```
sudo apt build-dep deadbeef-dmo
sudo apt build-dep audacious
wget https://netix.dl.sourceforge.net/project/deadbeef/travis/linux/1.8.8/deadbeef-1.8.8.tar.bz2
unpack
cd deadbeef-1.8.8
export CC=gcc
export CXX=g++
autoreconf -fi
./configure --help
```
`LDFLAGS='-Wl,-rpath,/opt/deadbeef188/lib,-rpath,/opt/deadbeef188/lib/deadbeef' ./configure --prefix=/opt/deadbeef188 --enable-silent-rules --enable-shared --disable-static --disable-staticlink --enable-portable=no --enable-pulse --disable-oss --disable-coreaudio --enable-gtk2 --disable-gtk3 --disable-lfm --disable-artwork-network --disable-artwork-imlib2 --enable-supereq --enable-rgscanner --enable-converter --enable-mp3 --enable-m3u --enable-flac --enable-src --disable-sid --disable-vtx --disable-sc68 --disable-psf --disable-gme --disable-vfs-curl --disable-vfs-zip --disable-adplug --disable-ffap --disable-notify --disable-wildmidi --disable-soundtouch --disable-mono2stereo --disable-mms --disable-shellexec --disable-shellexecui --disable-dumb --disable-shn --disable-tta --disable-dca --enable-opus --libdir=/opt/deadbeef188/lib`
```
make
sudo make install
mkdir -p ~/.local/share/applications
cp /opt/deadbeef188/share/applications/deadbeef.desktop ~/.local/share/applications/deadbeef.desktop

deadbeef.desktop
Exec=env XDG_DATA_DIRS=/usr/share:/usr/local/share:/opt/deadbeef188/share /opt/deadbeef188/bin/deadbeef %F
Icon=/opt/deadbeef188/share/icons/hicolor/128x128/apps/deadbeef.png
```
XDG_DATA_DIRS is optional.

**Comment by lumixdeee** [comment]
Succeeded in making the install, but when I run the deadbeef executable from terminal, the last message:

`gui plugin has quit; waiting for mainloop thread to finish`

htop shows deadbeef running, but no gui! Any ideas?

**Comment by Oleksiy-Yakovenko** [comment]
@minuxlintebiandedition  seems like you compiled deadbeef, but didn't check the list of plugins when running ./configure, and most likely you missed some dependencies - so the UI component was not compiled, and most likely many other components weren't compiled either.

**Reply by lumixdeee** [reply]
plugin summary says yes for both gtk plugins! says no for mp3, vorbis, flac etc

**Comment by Oleksiy-Yakovenko** [comment]
In this case, the UI should work. Can you post the full console log when running deadbeef? This should help to understand the problem.

**Comment by lumixdeee** [comment]
```starting deadbeef 0.7.2
server_start
searching for GUI plugins in /home/phifox/.local/lib32/deadbeef
searching for GUI plugins in /home/phifox/.local/lib/deadbeef
searching for GUI plugins in /usr/local/lib/deadbeef
load_plugin_dir /usr/local/lib/deadbeef: scandir found 49 files
load gui plugin
selected GUI plugin not found or failed to load, trying to find another GUI plugin
loading plugins from /home/phifox/.local/lib32/deadbeef
loading plugins from /home/phifox/.local/lib/deadbeef
loading plugins from /usr/local/lib/deadbeef
load_plugin_dir /usr/local/lib/deadbeef: scandir found 49 files
loading plugin /usr/local/lib/deadbeef/adplug.so
loading plugin /usr/local/lib/deadbeef/alac.so
loading plugin /usr/local/lib/deadbeef/dca.so
loading plugin /usr/local/lib/deadbeef/ddb_ao.so
loading plugin /usr/local/lib/deadbeef/ddb_dumb.so
loading plugin /usr/local/lib/deadbeef/ddb_mono2stereo.so
loading plugin /usr/local/lib/deadbeef/ddb_shn.so
loading plugin /usr/local/lib/deadbeef/flac.so
loading plugin /usr/local/lib/deadbeef/gme.so
loading plugin /usr/local/lib/deadbeef/in_sc68.so
loading plugin /usr/local/lib/deadbeef/m3u.so
loading plugin /usr/local/lib/deadbeef/mms.so
loading plugin /usr/local/lib/deadbeef/mp3.so
loading plugin /usr/local/lib/deadbeef/musepack.so
loading plugin /usr/local/lib/deadbeef/nullout.so
loading plugin /usr/local/lib/deadbeef/oss.so
loading plugin /usr/local/lib/deadbeef/shellexec.so
loading plugin /usr/local/lib/deadbeef/sid.so
loading plugin /usr/local/lib/deadbeef/supereq.so
loading plugin /usr/local/lib/deadbeef/tta.so
loading plugin /usr/local/lib/deadbeef/vtx.so
loading plugin /usr/local/lib/deadbeef/wildmidi.so
loading plugin /usr/local/lib/deadbeef/wma.so
selected output plugin: Null output plugin
INFO: loading playlist Default
INFO: from file /home/phifox/.config/deadbeef/playlists/0.dbpl
gui plugin has quit; waiting for mainloop thread to finish```


This was on machine 1, a current era stable debian.  Thought it might work better in it's own era so I set up a new machine, machine 2, based on oldoldstable, lmde cindy, with deb-multimedia repos and over a thousand extra multimedia dev related files. Can now get most of the configure script to 'yes', but 'sudo make install' this time resulted in a deadbeef folder filled with plugin files, but no deadbeef executable.
Would machine 1 or 2 be better  for this?

**Comment by Oleksiy-Yakovenko** [comment]
@minuxlintebiandedition this log shows that there's no UI plugin, and no Output plugin. This build won't work for playing music -- you need to look at your build / install process, and figure out why GTKUI, pulse and alsa plugins failed to build or install.

**Comment by lumixdeee** [comment]
Now I have all yes to all plugin checks except CoreAudio. I'm needing to 'sudo make' instead of 'make' because otherwise it fails with an error can't delete lastfm.so.

End of sudo make install output is  now :

`93 warnings generated.

  CC     ddb_gui_GTK3_la-tfimport.lo

  OBJCLD ddb_gui_GTK3.la

libtool: link: unable to infer tagged configuration
libtool: link: specify a tag with `--tag'
make[1]: *** [Makefile:715: ddb_gui_GTK3.la] Error 1
make[1]: Leaving directory ;/home/phifox/Templates/deadbeef-static/deadbeef-0.7.2/plugins/gtkui;
make: *** [Makefile:769: install-recursive] Error 1`

**Comment by Oleksiy-Yakovenko** [comment]
Is there any reason why you are trying to compile deadbeef 0.7.2 for 32 bit on your own? You can simply download it from the archive at [this link](https://sourceforge.net/projects/deadbeef/files/Archive/deadbeef-static_0.7.2-2_i686.tar.bz2/download)

**Reply by lumixdeee** [reply]
Was trying to get it in .deb format so it can be easily installed with dpkg or gdebi
I've downloaded that link before, it works but I had trouble getting it to work with the menu and the icons etc.

I wanted it without the gtk2 dependency because there isn't enough space for un-needed stuff on the asus eee pc 2g surf.

out if interest, if you try to compile it, does it work?
I don't understand how I managed to get an executable with no gui earlier, but now I can't even get the executable, despite having nearly all 'yes' in config report.
Does the bison (or Bison?) version matter?

**Comment by Oleksiy-Yakovenko** [comment]
> out if interest, if you try to compile it, does it work?

surely it does -- how do you think I compiled the build which I linked above?

**Comment by Oleksiy-Yakovenko** [comment]
> Does the bison (or Bison?) version matter?

deadbeef doesn't use bison, so no -- it should not matter. it's probably something that autoconf checks by default.
Anyway..  judging by the problems you are writing in the previous messages, I would recommend to start from scratch.
Delete whatever you downloaded, unpack freshly, then repeat the instructions from the INSTALL file.
Make sure you DO NOT run anything as root (DO NOT use `sudo` for anything, except for `make install`)

**Comment by lumixdeee** [comment]
Successful milestone achieved! Have managed to compile and install a deadbeef executable that has a gui and plays an .ogg file but has no dependency to gtk2.0
However, it isn't the static version, my error, so trying again now. Using checkinstall package to turn the install into a .deb format.
Rather than deleting everything, set up machine 4 : LMDE 5 with dmo repos added. Removed bloat then Followed [artenax](https://github.com/artenax)'s instructions. Success!

Making static version now, then will test install on a fresh machine with no prior deadbeef installation or any development tools or langs.

**Comment by lumixdeee** [comment]
More success: Have now successfully complied a working version, in static .deb form. It appears to play music gaplessly, which is the obvious reason to use deadbeef.
However, my static version is using 45.8 MB on disk, as opposed to the 15.7 MB used by version 6.x.x of deadbeef-static. I wonder why?
How can I make mine 15 MB in size?
I only ever play mp3, ogg, opus, flac, aac, wav, and wma and use the drag and drop, or open file/folder or open locations such as radio streams. How small can deadbeef-static be?
Is it large because it is now 2022 and everything is larger now? Would it be smaller if I compiled it on an oldoldstable machine?

Update: It seems top be a problem with my install creation. In my version, there is a textfile.la and an archive.a for each widget.so
The archive files are large, totalling 33.9 MB. Removing the archive.a files reduces the size to approx 12 MB, This is good news but also a worry because if it is smaller than the previous version, what has changed? Is something missing.
Jobs for me for the next year : Test, test, and test.

**Reply by Oleksiy-Yakovenko** [reply]
Official build sizes:
1.9.2: 18.7MB unpacked, 8.3MB packed.
0.7.2: 18.5MB unpacked, 8.1MB packed

So your resulting build size is not too far off, but yeah it's not the same number, which may indicate that you're building deadbeef in some different way than how official builds are made.

**Comment by lumixdeee** [comment]
Rather hasty yesterday, tested with .ogg and .flac, 
Today tried .mp3 there's no mp3 plugin working, but can play mp3 through the ffmpeg plugin ie non-critical
Can't get it to play internet radio, which is critical! How to ensure this when compiling?

The numbers you give there are for deadbeef not deadbeef-static, so will be smaller!
I'd assumed you'd tried compiling and packaging deadbeef-static 0.7.1 at some point, though not released?

**Reply by Oleksiy-Yakovenko** [reply]
> Can't get it to play internet radio, which is critical! How to ensure this when compiling?

It depends on the radio ... basically, you need the transport plugins -- `vfs_curl`, `mms`, as well as codecs -- vorbis, aac, mp3.

**Reply by Oleksiy-Yakovenko** [reply]
> The numbers you give there are for deadbeef not deadbeef-static, so will be smaller!

The numbers I provided are from the static builds of deadbeef, as published on the deadbeef website.

> I'd assumed you'd tried compiling and packaging deadbeef-static 0.7.1 at some point, though not released?

Not sure what you mean by this question. All releases were compiled, packaged and released at some point.
You can still download the sources and the builds from the archive, down to release 0.1.0 from 2009.

https://sourceforge.net/projects/deadbeef/files/Archive/

(although we didn't have official builds until release 0.4.3, and distro packages before 1.8.0 were not archived)

**Comment by lumixdeee** [comment]
Hello! Succeeded in making a version that does everything, on an oldoldstable box, and this worked on another oldoldstable box, after installing three dependancies.
Then tried this build on the target machine, a debian stable with very minimal software (entire system 1.5GB). Found my static build was missing many deps.
Then installed deadbeef 1.8.8 on the target box, removed deadbeef, installed my copy of deadbeef-static, found it works as well as it did on the other machine! However, this isn't what I was expecting from deadbeef-static.
Also, found deadbeef-static 0.6 doesn't work on current stable minimal box.
It's been interesting trying to compile, but I don't think there's any way to do what I originally wanted.
I never did the ./autogen.sh part. What does it do?

**Reply by Oleksiy-Yakovenko** [reply]
It's quite hard to understand what are the problems without knowing you exact steps and your expectations.
Deadbeef 0.6 is from 2014 -- that's 8 years ago. Many things have changed since then, which may have caused incompatibilities, which later got fixed in newer deadbeef releases.
The latest official build is expected to work on most mainstream linux distributions, as old as 2014 (Ubuntu 14.04).
Whatever you build locally is your own responsibility.

> I never did the ./autogen.sh part. What does it do?

If you never ran it -- your probably don't need it. Although you can easily find out what it does by opening it in a text editor :)
